code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 		lit = 0
 		update_brightness()
 		visible_message("<span class='warning'>Without warning, the flame on \the [src] suddenly goes out in a weak fashion.</span>")
+
+/obj/item/weapon/vape
+	name = "E-Cigarette"
+	desc = "More commonly referred to as a mouth fedora."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "vape"
+	var/datum/effect/effect/system/smoke_spread/smoke = new()
```

Comment:
Why isn't Travis building this? The outcome of the compile would answer the question.